lib/parser.mly

```diff
@@ -1,599 +0,0 @@
-(** The parser for Stan. A Menhir file. *)
-
-%{
-open Ast
-open Debug
-
-let initialize_expr_meta startpos endpos =
-  {expr_untyped_meta_loc= Location (startpos, endpos)}
-
-let initialize_stmt_meta startpos endpos =
-  {stmt_untyped_meta_loc= Location (startpos, endpos)}
-
-(* Takes a sized_basic_type and a list of sizes and repeatedly applies then
-   SArray constructor, taking sizes off the list *)
-let reducearray (sbt, l) =
```

Comment:
There are some List module functions (maybe more in Core_kernel) to do various kinds of folds and reduces

Reply:
Done.